src/request_body_processor/multipart.cc

```diff
@@ -935,6 +935,7 @@ int Multipart::process_part_header(std::string *error, int offset) {
                 "Multipart: Added part header \"" + header_name \
                 + "\" \"" + header_value + "\".");
             if (len_without_termination > 0) {
+                m_buf[len_without_termination] = '\0';
```

Comment:
Your diagnosis appears to be correct, but I don't think we want to solve it this way.
 
m_buf is the unparsed buffer of data. It's typically undesirable to overwrite the pre-parsed buffer when extracting values from it. Although I don't see any tangible issues that would occur with the current code base, it may set us up for a problem later if someone does not realize that the original buffer has been altered.

Perhaps consider instead merely altering the assignment line to use the overloaded function that includes length? E.g.

m_mpp->m_last_header_line.assign(m_buf, len_without_termination);

Since it is the extracted value that needs the null-terminator byte, that's the ideal place to add it.

Reply:
> Perhaps consider instead merely altering the assignment line to use the overloaded function that includes length? > E.g. 
> 
> m_mpp->m_last_header_line.assign(m_buf, len_without_termination);

You are right, I'm going to review the PR and change this.

Reply:
Thanks for the feedback, it's done - see 6623c0ae.